skyportal/models/analysis.py

```diff
@@ -189,17 +199,148 @@ def authinfo(self, value):
 
 
+class DictNumpyEncoder(json.JSONEncoder):
+    def default(self, obj):
+        if isinstance(obj, np.ndarray):
+            return obj.tolist()
+        return json.JSONEncoder.default(self, obj)
+
+
 class AnalysisMixin:
     def calc_hash(self):
-        md5_hash = hashlib.md5()
-        md5_hash.update(self._data.to_netcdf())
-        self.hash = md5_hash.hexdigest()
+        self.hash = joblib.hash(self.filename)
+
+    @hybrid_property
+    def has_inference_data(self):
+        return self.data.get('inference_data', None) is not None
+
+    @hybrid_property
+    def has_plot_data(self):
+        return self.data.get('plots', None) is not None
+
+    @hybrid_property
+    def number_of_analysis_plots(self):
+        if not self.has_plot_data:
+            return 0
+        else:
+            return len(self.data.get('plots', []))
+
+    @hybrid_property
+    def has_results_data(self):
+        return self.data.get('results', None) is not None
+
+    def serialize_results_data(self):
+        """
+        return the results data as a dictonary, even if it
+        contains some numpy arrays
+        """
+        if not self.has_results_data:
+            return {}
+        results = self.data.get('results', {"format": "json", "data": {}})
+        if not isinstance(results, dict):
+            return {}
+
+        if results.get('format', None) == "json":
+            return results.get('data', {})
+        elif results.get('format', None) == "joblib":
+            try:
+                buf = io.BytesIO()
+                buf.write(base64.b64decode(results.get('data', None)))
+                buf.seek(0)
+                data = joblib.load(buf)
+                jsons = json.dumps(data, cls=DictNumpyEncoder)
+            except Exception as e:
+                log(f"Error serializing results data: {e}")
+                jsons = "{}"
+            return json.loads(jsons)
+        else:
+            return {}
+
+    def get_analysis_plot(self, plot_number=0):
+        if not self.has_plot_data:
+            return None
+
+        if plot_number < 0 or plot_number >= self.number_of_analysis_plots:
+            return None
+
+        plot = self.data.get('plots')[plot_number]
+        try:
+            format = plot["format"]
+        except Exception as e:
+            format = "png"
+            log(f"Warning: missing format in plot, assuming png {e}")
+
+        buf = io.BytesIO()
+        buf.write(base64.b64decode(plot['data']))
+        buf.seek(0)
+
+        return {"plot_data": buf, "plot_type": format}
+
+    def generate_corner_plot(self, **plot_kwargs):
+        """Generate a corner plot of the posterior from the inference data."""
+
+        if not self.has_inference_data:
+            return None
+
+        # we could add different formats here in the future
+        # but for now we only support netcdf4 formats
+        if self.data['inference_data']["format"] != "netcdf4":
+            return None
```

Comment:
Shouldn't this be an error of some sort?

Reply:
Yes...and now I catch it in the API handler.